internal/services/containers/container_registry_task_resource.go

```diff
@@ -897,9 +897,18 @@ func (r ContainerRegistryTaskResource) Update() sdk.ResourceFunc {
 			if metadata.ResourceData.HasChange("docker_step") || metadata.ResourceData.HasChange("file_step") || metadata.ResourceData.HasChange("encoded_step") {
 				existing.TaskProperties.Step = expandRegistryTaskStep(model)
 			}
+
 			if metadata.ResourceData.HasChange("base_image_trigger") || metadata.ResourceData.HasChange("source_trigger") || metadata.ResourceData.HasChange("timer_trigger") {
 				existing.TaskProperties.Trigger = expandRegistryTaskTrigger(model)
 			}
+
+			if existing.TaskProperties.Trigger != nil {
+				if !metadata.ResourceData.HasChange("source_triggers") {
+					// For update that is not affecting source_triggers, we need to patch the source_triggers to include the properties missing in the response of GET.
+					existing.TaskProperties.Trigger.SourceTriggers = patchRegistryTaskTriggerSourceTrigger(existing.TaskProperties.Trigger.SourceTriggers, model)
```

Comment:
fwiw you can remove all of the de-references (and the nil-check, since it's covered above) below if we do it here instead:

```suggestion
					existing.TaskProperties.Trigger.SourceTriggers = patchRegistryTaskTriggerSourceTrigger(*existing.TaskProperties.Trigger.SourceTriggers, model)
```